From 3693332c5ee2c38464e9843dfcf7c626747cb3a3 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Thu, 14 Feb 2008 16:16:36 +0200
Subject: [PATCH] sendxmpp: Add CPAN check

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 sendxmpp |   28 ++++++++++++++++++++++++----
 1 files changed, 24 insertions(+), 4 deletions(-)

diff --git a/sendxmpp b/sendxmpp
index 3959ffb..8ef1180 100644
--- a/sendxmpp
+++ b/sendxmpp
@@ -19,13 +19,33 @@ if 0; # not running under some shell
 # $Platon: sendxmpp/sendxmpp,v 1.13 2007-09-10 19:08:35 rajo Exp $
 # $Id: $
 
-use Authen::SASL qw(Perl); # authentication broken if Authen::SASL::Cyrus module installed
-use Net::XMPP;
+# use Authen::SASL qw(Perl); # authentication broken if Authen::SASL::Cyrus module installed
 use Getopt::Long;
 use strict;
 
-use open ':utf8';                                                                                                                                                                    
-use open ':std';                                                                                                                                                                     
+# CPAN MODULES
+
+BEGIN
+{
+    my @list;
+
+    for ( 'Net::XMPP' )
+    {
+        eval "use $_;" ;
+        push @list, $_   if $@;  # Eval error
+    }
+
+    if ( @list )
+    {
+        die q([FATAL] Perl CPAN modules are needed. Please install them with:
+
+perl -MCPAN -e 'install $_ for @ARGV' ) . qq(@list\n);
+
+    }
+}
+
+use open ':utf8';
+use open ':std';
 
 # subroutines decls
 sub xmpp_login($$$$$$$$);
-- 
1.5.3.8

